icu4c/source/i18n/zonemeta.cpp

```diff
@@ -290,13 +290,9 @@ ZoneMeta::getCanonicalCLDRID(const UnicodeString &tzid, UErrorCode& status) {
     UResourceBundle *rb = ures_getByKey(top, gTypeMapTag, NULL, &tmpStatus);
     ures_getByKey(rb, gTimezoneTag, rb, &tmpStatus);
     ures_getByKey(rb, id, rb, &tmpStatus);
-    int32_t bcp47StrLen = 0;
-    ures_getString(rb, &bcp47StrLen, &tmpStatus);
```

Comment:
Revert the addition of code to exclude 4-character metazone short IDs added in the previous integration PR #1843, since those IDs are now moved to a different table